tests/unit/metadata/usermetadatasize/usermetadatasize.js

```diff
@@ -0,0 +1,42 @@
+const assert = require('assert');
+const { invalidObjectUserMetadataHeader } = require('../../../../constants');
+const genMaxSizeMetaHeaders = require(
+    '../../../functional/aws-node-sdk/lib/utility/genMaxSizeMetaHeaders');
+const checkUserMetadataSize
+    = require('../../../../lib/api/apiUtils/object/checkUserMetadataSize');
+
+const userMetadataKey = 'x-amz-meta-';
+const metadata = {};
+
+let userMetadataKeys = 0;
+
+describe('Check user metadata size', () => {
+    before('Set up metadata', () => {
+        const md = genMaxSizeMetaHeaders();
+        Object.keys(md).forEach(key => {
+            metadata[`${userMetadataKey}${key}`] = md[key];
+        });
+        userMetadataKeys = Object.keys(metadata)
+            .filter(key => key.startsWith(userMetadataKey)).length;
+    });
+
+    it('Should return user metadata when the size is within limits', () => {
+        const responseMetadata = checkUserMetadataSize(metadata);
+        assert.notEqual(responseMetadata, undefined);
```

Comment:
Should we also check that no user metadata has been removed and no prescribed/custom metadata has been added?

Reply:
Yes. We should. I will correct the test. Thanks.